### PR TITLE
feat(ci): Add release-please automation with VERSION

### PR DESCRIPTION
**Pull request by petejohanson**
* Add VERSION file that Zephyr will pick up
* Add release-please automation
* Updated docs on commits messages, and PR process
* Updated PR template to match.

---

Release-As: 0.1.0

### CODE REVIEW

**Comment by petejohanson** [comment]
Tested all the various workflows in my ZMK fork. I've added the necessary secret/token (`ZMK_RELEASE_PLEASE_TOKEN`) for this already to the settings for this repo.

File: docs/docs/development/contributing/pull-requests.md [hunk 1]
```diff
@@ -68,3 +83,22 @@ The following scopes are frequently used to further clarify the scope of the cha
 - `core` -- changes to any other area of our core code
 - `boards` -- changes to the in-tree boards
 - `shields` -- changes to the in-tree shields
+
+## Opening a PR
+
+Create a PR by visiting https://github.com/zmkfirmware/zmk/pulls and clicking "New pull request" and selecting your branch. GitHub should auto-populate a start of a description/body to your PR, but please make sure to supplement that with additional details about the change by completing the sections in the PR template.
```
Comment (caksoylar) [inline docs/docs/development/contributing/pull-requests.md:89]:
```suggestion
Create a PR by visiting https://github.com/zmkfirmware/zmk/pulls and clicking "New pull request" and selecting your branch. GitHub should auto-populate a start of a description/body to your PR, but please make sure to supplement that with additional details about the change and make sure the check-list is complete in the PR template.
```

File: docs/docs/development/contributing/pull-requests.md [hunk 2]
```diff
@@ -1,12 +1,27 @@
...
+
+## Clean Commit History
+
+Before opening a PR, ensure your branch has a clean commit history, which allows our release automation to generate clean changelogs, and allows easier understanding of our commit history when investigating regressions, etc.
+
+A "clean" commit history satisfies the following:
+
+- Commit messages follow our [commit message conventions](#commit-messages).
+- The sequence of commits should be well organized, with discrete commits used to break any combined work into smaller, cohesive changes if the scope of a given PR/change is larger. This allows for easier code review. You can usually accomplish this by amending (e.g. `git add foo && git commit --amend`) or by interactive rebasing (e.g. `git rebase -i upstream/main`) and squashing/rewording commits.
+- Do not use merge commits to catch up with ZMK `main`, instead using rebasing to address any issues with divergence (e.g. `git rebase upstream/main`)
```
Comment (caksoylar) [inline docs/docs/development/contributing/pull-requests.md:16]:
```suggestion
- Do not use merge commits to catch up with ZMK `main`, instead using rebasing to address any issues with divergence (e.g. `git rebase upstream/main`).
```

File: docs/docs/development/contributing/pull-requests.md [hunk 3]
```diff
@@ -1,12 +1,27 @@
 ---
-title: Commit Messages
+title: Pull Requests
+sidebar_title: PRs
```
Comment (caksoylar) [inline docs/docs/development/contributing/pull-requests.md:3]:
```suggestion
sidebar_title: Pull Requests
```
"Pull Requests" doesn't seem too long to me for the sidebar.

**Review by caksoylar — APPROVED** [review]

**Pull request merged**